File: labs/03/YearCodes.java
/* YearCodes.java is a driver for function yearCode().
 * Student: Jacob Brink
 * Date: 2/20/2020
 * Lab: 03
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 ****************************************************************/

import java.util.Scanner;

public class YearCodes {

    
   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your academic year: "); // prompt

      String year = keyboard.next();      // read year
      System.out.println(yearCode(year)); // display its code
   }

   /***************************************************************
    * yearCode() converts an academic year into its integer code.
    *
    * Receive: year, a string.
    * PRE: yearName == one of {freshman, sophomore, junior, or senior}.
    * Return: the integer code corresponding to year.
    */
    
    public static int yearCode(String year)
    {
	if (year.equals("freshman"))
	    return 1;
	else if (year.equals("sophomore"))
	    return 2;
	else if (year.equals("junior"))
	    return 3;
	else if (year.equals("senior"))
	    return 4;
	else
	    return 0;
    }
    
}

File: labs/03/year_codes.rb
#! /usr/bin/ruby
# year_codes.rb translates an academic year into a numeric code
# Begun by: Dr. Nelesen, for CS 214 at Calvin College
# Completed by: Jacob Brink
# Lab: 03
# Date: 2/21/2020
################################################################

# Input:  The name of an academic year
# Precondition: The academic year is a string with a value of freshman, 
#   sophomore, junior or senior
# Output: The corresponding integer code for the given academic year

def yearCode(year)
  if year =~ /freshman/ then
    1
  elsif year =~ /sophomore/ then
    2
  else
    if year =~ /junior/ then
      3
    elsif year =~ /senior/ then
      4
    else
      0
    end
  end
    
end


if __FILE__ == $0
   print "Enter the year: "
   year = gets
   print "Numeric code is: "
   puts yearCode(year)
end

